Src/FluentAssertions/Formatting/FormattedObjectGraph.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Formatting
+{
+    /// <summary>
+    /// This class is used by the <see cref="Formatter"/> class to collect all the output of the (nested calls of a) <see cref="IValueFormatter"/> into
+    /// a the final representation.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="FormattedObjectGraph"/> will ensure that the number of lines will be limited
+    /// to the maximum lines provided through its constructor. It will throw
+    /// a <see cref="MaxLinesExceededException"/> when the maximum has been reached.
```

Comment:
🔧 (as we often do not reach the limit)
```suggestion
    /// a <see cref="MaxLinesExceededException"/> if the maximum has been reached.
```